aws-s3/spring-cloud-starter-stream-source-s3/src/main/java/org/springframework/cloud/stream/app/s3/source/AmazonS3SourceConfiguration.java

```diff
@@ -55,6 +57,11 @@
 	private ResourceIdResolver resourceIdResolver;
 
 	@Bean
+	public TriggerProperties triggerProperties() {
+		return new TriggerPropertiesMaxMessagesDefaultUnlimited();
```

Comment:
Seems for me `@EnableConfigurationProperties` does exactly what we need with population `@ConfigurationProperties` classes as beans.
Therefore `TriggerConfiguration` is still able to be `@Autiwired` with `TriggerProperties` impl.
Am I missing anything there around ?